demos/pet-demo/src/components/__isograph/Pet/PetCheckinsCard/__refetch__0.ts

```diff
@@ -62,6 +64,8 @@ const normalizationAst: NormalizationAst = [
 const artifact: RefetchQueryNormalizationArtifact = {
   kind: "RefetchQuery",
   queryText,
+  concreteType: "Mutation",
+  queryType: "Query",
```

Comment:
is this correct?

Reply:
so the `queryType` is generated on every entrypoint in case of an edge case where we need to normalize nested `Query`

it would be much better if there was single place to generate this,
the best I can think of right now is to generate this in `iso.ts`


```tsx
import { _createIsographEnvironment } from "@isograph/react"

export function createIsographEnvironment(...args){
  return _createIsographEnvironment(...args, {
    queryType: "Query"
  })
}

```

Reply:
Sorry, I don't really remember what the edge case is. If we have `foo: Query`, what goes wrong without this?

Reply:
In this case we want `getDataIdOfNetworkResponse` to return `__ROOT`, so the object gets normalized in `store["Query"]["__ROOT"]`,
Right now it will calculate a different key `ParentObject:123.foo` and normalize in `store["Query"]["ParentObject:123.foo"]`,
Every time we read, we will start from `store["Query"]["__ROOT"]` so we will miss those records

Reply:
ohhhh, hm. I think the optimal solution here is to have a directive on the type that we parse `@globalSingleton` that controls this behavior. We can skip this for now. If someone runs into this issue, we can reprioritize